acme/acme/messages.py

```diff
@@ -201,60 +209,59 @@ class Meta(jose.JSONObjectWithFields):
         caa_identities = jose.Field('caaIdentities', omitempty=True)
         external_account_required = jose.Field('externalAccountRequired', omitempty=True)
 
-        def __init__(self, **kwargs):
+        def __init__(self, **kwargs: Any) -> None:
             kwargs = {self._internal_name(k): v for k, v in kwargs.items()}
             super().__init__(**kwargs)
 
         @property
-        def terms_of_service(self):
+        def terms_of_service(self) -> str:
             """URL for the CA TOS"""
             return self._terms_of_service or self._terms_of_service_v2
 
-        def __iter__(self):
+        def __iter__(self) -> Iterator[str]:
             # When iterating over fields, use the external name 'terms_of_service' instead of
             # the internal '_terms_of_service'.
             for name in super().__iter__():
                 yield name[1:] if name == '_terms_of_service' else name
 
-        def _internal_name(self, name):
+        def _internal_name(self, name: str) -> str:
             return '_' + name if name == 'terms_of_service' else name
 
-
     @classmethod
-    def _canon_key(cls, key):
+    def _canon_key(cls, key: str) -> str:
         return getattr(key, 'resource_type', key)
 
     @classmethod
-    def register(cls, resource_body_cls: Type[Any]) -> Type[Any]:
+    def register(cls, resource_body_cls: Type['Directory']) -> Type['Directory']:
         """Register resource."""
         resource_type = resource_body_cls.resource_type
         assert resource_type not in cls._REGISTERED_TYPES
         cls._REGISTERED_TYPES[resource_type] = resource_body_cls
         return resource_body_cls
 
-    def __init__(self, jobj):
+    def __init__(self, jobj: Dict[str, Any]) -> None:
         canon_jobj = util.map_keys(jobj, self._canon_key)
         # TODO: check that everything is an absolute URL; acme-spec is
         # not clear on that
         self._jobj = canon_jobj
 
-    def __getattr__(self, name):
+    def __getattr__(self, name: str) -> Any:
         try:
             return self[name.replace('_', '-')]
         except KeyError as error:
             raise AttributeError(str(error))
 
-    def __getitem__(self, name):
+    def __getitem__(self, name: str) -> Any:
         try:
             return self._jobj[self._canon_key(name)]
         except KeyError:
             raise KeyError('Directory field "' + self._canon_key(name) + '" not found')
 
-    def to_partial_json(self):
+    def to_partial_json(self) -> Dict[str, Any]:
         return self._jobj
 
     @classmethod
-    def from_json(cls, jobj):
+    def from_json(cls, jobj: Dict[str, Any]) -> 'Directory':
         jobj['meta'] = cls.Meta.from_json(jobj.pop('meta', {}))
         return cls(jobj)
```

Comment:
(Nit alert) this stands out to me. We changed this to `Mapping` in most other places, except it'd have to be `MutableMapping` here because of the way the method is implemented.

There's a few `Dict` arguments remaining in this file, but I will leave it to your judgement as to whether they should change or not.